components/camel-graphql/src/main/java/org/apache/camel/component/graphql/GraphqlComponent.java

```diff
@@ -0,0 +1,18 @@
+package org.apache.camel.component.graphql;
+
+import java.util.Map;
+
+import org.apache.camel.Endpoint;
+import org.apache.camel.spi.annotations.Component;
+import org.apache.camel.support.DefaultComponent;
+
+@Component("graphql")
+public class GraphqlComponent extends DefaultComponent {
+
+    @Override
+    protected Endpoint createEndpoint(String uri, String remaining, Map<String, Object> parameters) throws Exception {
+        GraphqlConfiguration configuration = new GraphqlConfiguration(remaining, parameters);
```

Comment:
I think would be better to follow other components on how to set configurations, meaning that to configuration properties mutable and let Camel do the job of binding the parameters via `setProperties(configuration, parameters);`.

Reply:
Can Camel convert multiple variables prefixed with "variable." to a Map<String, Object>?

Reply:
Done